src/algebra/lie/free.lean

```diff
@@ -98,38 +98,36 @@ instance : has_scalar R (free_lie_algebra R X) :=
 instance : has_add (free_lie_algebra R X) :=
 { add := quot.map₂ (+) rel.add_left rel.add_right, }
 
-instance : add_comm_group (free_lie_algebra R X) :=
+instance : add_comm_monoid (free_lie_algebra R X) :=
 { add_comm       := by { rintros ⟨a⟩ ⟨b⟩, change quot.mk _ _ = quot.mk _ _, rw add_comm, },
   add_assoc      := by { rintros ⟨a⟩ ⟨b⟩ ⟨c⟩, change quot.mk _ _ = quot.mk _ _, rw add_assoc, },
   zero           := quot.mk _ 0,
   zero_add       := by { rintros ⟨a⟩, change quot.mk _ _ = _, rw zero_add, },
   add_zero       := by { rintros ⟨a⟩, change quot.mk _ _ = _, rw add_zero, },
-  neg            := λ x, (-1 : R) • x,
-  sub            := λ x y, x + ((-1 : R) • y),
-  sub_eq_add_neg := λ x y, rfl,
-  add_left_neg   :=
-    by { rintros ⟨a⟩, change quot.mk _ _ = _, erw [neg_smul, one_smul, add_left_neg], refl, },
-  ..(infer_instance : has_add _)}
+  .. (infer_instance : has_add _), }
+
+instance : module R (free_lie_algebra R X) :=
+{ one_smul  := by { rintros ⟨a⟩, change quot.mk _ _ = quot.mk _ _, rw one_smul, },
+  mul_smul  := by { rintros t₁ t₂ ⟨a⟩, change quot.mk _ _ = quot.mk _ _, rw mul_smul, },
+  add_smul  := by { rintros t₁ t₂ ⟨a⟩, change quot.mk _ _ = quot.mk _ _, rw add_smul, },
+  smul_add  := by { rintros t ⟨a⟩ ⟨b⟩, change quot.mk _ _ = quot.mk _ _, rw smul_add, },
+  zero_smul := by { rintros ⟨a⟩, change quot.mk _ _ = quot.mk _ _, rw zero_smul, },
+  smul_zero := λ t, by { change quot.mk _ _ = quot.mk _ _, rw smul_zero, }, }
+
+instance : add_comm_group (free_lie_algebra R X) :=
+module.add_comm_monoid_to_add_comm_group R
```

Comment:
I think this definition should go in `free_non_unital_non_assoc_algebra`, not here.

Reply:
Then in this file you can define
```lean
lemma rel.neg (a b : lib R X) (h : rel R X a b) : rel R X (-a) (-b) :=
h.smul (-1) _ _
```
and
```lean
instance : has_neg (free_lie_algebra R X) :=
{ neg := quot.map has_neg.neg rel.neg, }
```

Reply:
I've just added https://github.com/leanprover-community/mathlib/pull/8163/commits/8f4e580f0450be77c9dcdb3804c4e426f275ce64 which fills the gap in `free_non_unital_non_assoc_algebra`.

Are you sure it's better to define the `has_neg` manually here and not just use `module.add_comm_monoid_to_add_comm_group` again?

Reply:
Ah I just went for it.

Reply:
> Are you sure it's better to define the has_neg manually here and not just use module.add_comm_monoid_to_add_comm_group again?

It matches what I did for the clifford algebra, so I think is sensible.